qiskit/circuit/quantumcircuit.py

```diff
@@ -807,6 +807,48 @@ def copy(self, name=None):
             cpy.name = name
         return cpy
 
+    def measure_active(self):
+        """Adds measurement gates to all non-idle qubits. Creates a new ClassicalRegister with
+        a size equal to the number of non-idle qubits being measured."""
+        from qiskit.converters.circuit_to_dag import circuit_to_dag
+        dag = circuit_to_dag(self)
+        qubits_to_measure = [qubit for qubit in self.qubits if qubit not in dag.idle_wires()]
+        new_creg = ClassicalRegister(len(qubits_to_measure), 'measure')
+        self.add_register(new_creg)
+        self.barrier()
+        self.measure(qubits_to_measure, new_creg)
+
+    def measure_all(self):
+        """Adds measurement gates to all qubits. Creates a new ClassicalRegister with a
+        size equal to the number of qubits being measured.
+        """
+        new_creg = ClassicalRegister(self.n_qubits, 'measure')
+        self.add_register(new_creg)
+        self.barrier()
+        self.measure(self.qubits, new_creg)
+
+    def remove_final_measures(self):
+        """Removes final measurement gates on all qubits if they are present.
+        Deletes the ClassicalRegister that was created to store the values from these measurements.
+        """
+        from qiskit.converters.circuit_to_dag import circuit_to_dag
+        cregs_to_remove = set()
+        for inst, _, cargs in reversed(self.data):
+            if inst.name == 'measure':
```

Comment:
can you make this general? it would be useful if this function could be invoked to remove final measurements on any circuit, not just those that have "measure" classical bits. I think this can be done more easily as a transpiler pass. In particular look into the `BarrierBeforeFinalMeasurement` pass which does similar analysis to find final measurements: https://github.com/Qiskit/qiskit-terra/blob/master/qiskit/transpiler/passes/mapping/barrier_before_final_measurements.py

Reply:
When this becomes a pass, this method on the QuantumCircuit can just internally convert to DAGCircuit, run the pass on it, and convert back to QuantumCircuit.

Reply:
So you're saying to create a new pass called remove_final_measurements, and then call that pass on the DAGCircuit of self in the QuantumCircuit method remove_final_measurements? I think that sounds cleaner than my approach. 

What do you mean by make the method more general. I thought my method would work on any circuit, not just ones that have ClassicalRegisters with the 'measure' name/prefix. Or am I misunderstanding.

Reply:
Yeah exactly, this is easier done on the DAG.

You're right I misread your code. It works well.